### PR TITLE
use localised duration for position mapping instead of global

### PR DESCRIPTION
**Pull request by dhruv2295**

### CODE REVIEW

File: lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt [hunk 1]
```diff
@@ -318,7 +320,10 @@ class DefaultThumbnailsEngine(
         if (stubs.firstOrNull()?.request?.sourceId() == source && positionUs == stubs.firstOrNull()?.positionUs) {
             return
         }
-        val locatedTimestampUs = SingleThumbnailRequest(positionUs).locate(timer.durationUs.video)[0]
+        val index = dataSources[TrackType.VIDEO].indexOfFirst { it.mediaId() == source }
```
Comment (sahilbajaj) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt]: Can we abstract it out?
May be an extension method in DataSources or directly within DataSources itself

Comment (sahilbajaj) [reply lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt]: We've used this in multiple places. Should help simply it.

**Review by sahilbajaj — CHANGES_REQUESTED** [review]
.

File: lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt [hunk 2]
```diff
@@ -259,21 +259,22 @@ class DefaultThumbnailsEngine(
...
 
         map.forEach { entry ->
-            val positions = entry.value.flatMap { request ->
-                val duration = timer.totalDurationUs
-                request.locate(duration).map { it to request }
-            }.sortedBy { it.first }
-            val index = dataSources[TrackType.VIDEO].indexOfFirst { it.mediaId() == entry.key }
-            if (index >= 0) {
+            val dataSource = getDataSourceForId(entry.key)
+            if (dataSource !=null) {
```
Comment (sahilbajaj) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt]: space - formatting

Comment (dhruv2295) [reply lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt]: done

**Review by sahilbajaj — APPROVED** [review]

**Pull request merged**